From b4c43ca0ec6bcb9da5b7ff8d27717bf971c71242 Mon Sep 17 00:00:00 2001
From: Feifan He <feifan@codeweavers.com>
Date: Fri, 2 Oct 2026 17:24:35 +0800
Subject: [PATCH 15/26] fix(dxmt): pass mipped width & height to clear encoder

Signed-off-by: Feifan He <feifan@codeweavers.com>

(cherry picked from commit f9beaea30b95c14e9ad76546e4a9bd9b726205c3)
Adapted by hand: the surrounding lines differ in v0.80.
---
 src/dxmt/dxmt_context.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/dxmt/dxmt_context.cpp b/src/dxmt/dxmt_context.cpp
index 6181fe4..1acf6ff 100644
--- a/src/dxmt/dxmt_context.cpp
+++ b/src/dxmt/dxmt_context.cpp
@@ -408,8 +408,8 @@ ArgumentEncodingContext::clearColor(Rc<Texture> &&texture, uint64_t viewId, unsi
   encoder_info->clear_dsv = 0;
   encoder_info->color = color;
   encoder_info->array_length = arrayLength;
-  encoder_info->width = texture->width();
-  encoder_info->height = texture->height();
+  encoder_info->width = texture->width(viewId);
+  encoder_info->height = texture->height(viewId);
   encoder_current = encoder_info;
 
   encoder_info->attachment = access(texture, viewId, ResourceAccess::Write);
@@ -432,8 +432,8 @@ ArgumentEncodingContext::clearDepthStencil(
   encoder_info->clear_dsv = flag & DepthStencilPlanarFlags(texture->pixelFormat());
   encoder_info->depth_stencil = {depth, stencil};
   encoder_info->array_length = arrayLength;
-  encoder_info->width = texture->width();
-  encoder_info->height = texture->height();
+  encoder_info->width = texture->width(viewId);
+  encoder_info->height = texture->height(viewId);
   encoder_current = encoder_info;
 
   encoder_info->attachment = access(texture, viewId, ResourceAccess::Write);
-- 
2.54.0 (Apple Git-157)


